sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala

```diff
@@ -1846,23 +1848,23 @@ class SessionCatalog(
   // -----------------
 
   /**
-   * Drop all existing databases (except "default"), tables, partitions and functions,
-   * and set the current database to "default".
+   * Drop all existing databases (except defaultDatabase), tables, partitions and functions,
+   * and set the current database to defaultDatabase.
    *
    * This is mainly used for tests.
    */
   def reset(): Unit = synchronized {
-    setCurrentDatabase(DEFAULT_DATABASE)
-    externalCatalog.setCurrentDatabase(DEFAULT_DATABASE)
-    listDatabases().filter(_ != DEFAULT_DATABASE).foreach { db =>
+    setCurrentDatabase(defaultDatabase)
+    externalCatalog.setCurrentDatabase(defaultDatabase)
+    listDatabases().filter(_ != defaultDatabase).foreach { db =>
```

Comment:
We can't drop DEFAULT database, so we should still filter by `DEFAULT_DATABASE` here.